crates/dojo-lang/src/compiler.rs

```diff
@@ -323,11 +333,35 @@ fn get_dojo_contract_artifacts(
         .iter()
         .filter(|name| !matches!(name.as_ref(), "world" | "executor" | "base"))
         .map(|name| {
+            let module_name = module_id.full_path(db);
+            let module_last_name = module_name.split("::").last().unwrap();
+
+            let reads = match SYSTEM_READS.lock().unwrap().get(module_last_name) {
```

Comment:
There isn't a way to handle this with aux data right?

Reply:
Yeah, Ori said Aux data should be the realm of proc macros...

If we want, we could also change things to either,
1. Do a full search on the whole module to find the `set!`/`get!` calls.
2. Do the same inside our Dojo contract proc macro.